From c5b9087393cc9ee37d435a9472e48a886260f7bc Mon Sep 17 00:00:00 2001
From: Andrew Tridgell <tridge@samba.org>
Date: Fri, 18 May 2012 21:02:27 +1000
Subject: [PATCH 2/2] capture: run as a SCHED_FIFO task

this lowers the frame loss from a PtGrey Chameleon on a pandaboard a lot
---
 libdc1394/dc1394/usb/capture.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libdc1394/dc1394/usb/capture.c b/libdc1394/dc1394/usb/capture.c
index 29aec6d..f2f4d00 100644
--- a/libdc1394/dc1394/usb/capture.c
+++ b/libdc1394/dc1394/usb/capture.c
@@ -26,6 +26,7 @@
 #include <inttypes.h>
 #include <pthread.h>
 #include <unistd.h>
+#include <sched.h>
 
 #include "usb/usb.h"
 
@@ -68,6 +69,11 @@ static void *
 capture_thread (void * arg)
 {
     platform_camera_t * craw = arg;
+    struct sched_param p;
+
+    // try to make the capture thread realtime
+    p.sched_priority = SCHED_FIFO;
+    sched_setscheduler(0, SCHED_FIFO, &p);
 
     dc1394_log_debug ("usb: Helper thread starting");
 
-- 
1.7.9.5

